dashboard/app/models/reference_guide.rb

```diff
@@ -0,0 +1,75 @@
+require 'fileutils'
+
+# == Schema Information
+#
+# Table name: reference_guides
+#
+#  id                :bigint           not null, primary key
+#  display_name      :string(255)
+#  key               :string(255)      not null
+#  course_version_id :integer          not null
+#  content           :text(65535)
+#  position          :integer          not null
+#  created_at        :datetime         not null
+#  updated_at        :datetime         not null
+#
+class ReferenceGuide < ApplicationRecord
+  belongs_to :course_version
+
+  validates_uniqueness_of :key, scope: :course_version_id
+
+  def serialize
+    {
+      display_name: display_name,
+      key: key,
+      course_version_key: course_version.key,
+      course_offering_key: course_version.course_offering.key,
+      content: content,
+      position: position
+    }
+  end
+
+  def write_serialization
+    return unless Rails.application.config.levelbuilder_mode
+    course_key = "#{course_version.course_offering.key}_#{course_version.key}"
+    file_path = Rails.root.join("config/reference_guides/#{course_key}/#{key}.json")
+    object_to_serialize = serialize
+    dirname = File.dirname(file_path)
+    unless File.directory?(dirname)
+      FileUtils.mkdir_p(dirname)
+    end
+    File.write(file_path, JSON.pretty_generate(object_to_serialize))
+  end
+
+  def self.seed_all
+    removed_records = all.pluck(:id)
```

Comment:
Does this work? It doesn't look like you are serializing id. I think this should probably be using key

Reply:
seed_record finds_or_initializes by key/course_version but then that method returns the id (local to that env) and we remove it from the set of all ids (also local to this env) here. If I used key I'd also have to use course_version_id. The ids won't be the same each seed, but that's okay because we're redoing this set of ids from scratch